internal/engine/task_buffer.go

```diff
@@ -0,0 +1,55 @@
+package engine
+
+import (
+	"sync"
+)
+
+// TaskBuffer represents a fixed-size buffer, it operates as a FIFO buffer
+type TaskBuffer struct {
```

Comment:
```suggestion
This is a new data structure, please add sufficient test cases.
```

Reply:
Comprehensive test cases have been added in `task_buffer_test.go` under directory`/engine`